examples/multimodal/wds.py

```diff
@@ -12,31 +13,25 @@
 
 meta_pq = (
     DataChain.from_parquet("gs://datachain-demo/datacomp-small/metadata/0020f*.parquet")
-    .filter(
```

Comment:
[Q] Can you use an inner join here? I need this to run within a reasonable amount of time for #199

Reply:
well... I actually plan to increase number of tar files from 1 to 60 in order to make it more realistic (match the data of the next steps 😅 )  which is an opposite direction.

Should we introduce a separate wds script for testing or create a small subset of WDS?

Reply:
> well... I actually plan to increase number of tar files from 1 to 60 in order to make it more realistic (match the data of the next steps 😅 ) which is an opposite direction.

How long will that take to run?

> Should we introduce a separate wds script for testing or create a small subset of WDS?

There are some separate WDS already. The question is whether or not we want to test all of the examples in the repository.

Reply:
The following runs in ~11m for me:

```python
from datachain import C, DataChain
from datachain.lib.webdataset import process_webdataset
from datachain.lib.webdataset_laion import WDSLaion, process_laion_meta
from datachain.sql.functions import path

wds_images = (
    DataChain.from_storage("gs://datachain-demo/datacomp-small/shards")
    .filter(C("file.name").glob("00000000.tar")) # check with single file
    .settings(cache=True)
    .gen(laion=process_webdataset(spec=WDSLaion), params="file")
)


wds_with_pq = (
    DataChain.from_parquet("gs://datachain-demo/datacomp-small/metadata/0020f*.parquet")
    .settings(cache=True)
    .merge(wds_images, on="uid", right_on="laion.json.uid", inner=True)
    .mutate(stem=path.file_stem(C("source.file.name")))
)

res = (
    DataChain.from_storage("gs://datachain-demo/datacomp-small/metadata/0020f*.npz")
    .settings(cache=True)
    .gen(emd=process_laion_meta)
    .mutate(stem=path.file_stem(C("emd.file.name")))
    .merge(
        wds_with_pq,
        on=["stem", "emd.index"],
        right_on=["stem", "source.index"],
        inner=True,
    )
    .save("wds")
)

res.show(5)
```

This should still scale, right?

Reply:
11 minutes looks good. 

Note, I increased the amount of data in the 1st step. Now it takes 1.5h in a laptop (see updated number in the description). It looks like you use different dataset.

Reply:
it seems valuable to keep it tests (we keep breaking it from my experience in Studio and CLI). would it be possible to parametrize and use a smaller storage / smaller files? (so that we can run in 1-2 mins on CI)?

Reply:
(I'm totally fine to drop it initially from the #199)

Reply:
@mattseddon Is your point that if we use an inner join, we can parametrize the query to scale up or down easily?

Reply:
> @mattseddon Is your point that if we use an inner join, we can parametrize the query to scale up or down easily?

Yes, that would be one of the benefits

Reply:
@dmpetrov WDYT about making it an inner join? That would prevent these kind of mismatches where there's a lot of unnecessary data joined.